torch/_refs/__init__.py

```diff
@@ -510,6 +512,38 @@ def _isnan(a: TensorLikeType) -> TensorLikeType:
 )
 
 
+@register_decomposition(torch.ops.aten.nan_to_num)
+@out_wrapper
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a,"),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
+)
+def nan_to_num(
+    a: TensorLikeType,
+    *,
+    nan: Optional[NumberType] = 0.0,
+    posinf: Optional[NumberType] = None,
+    neginf: Optional[NumberType] = None,
+) -> TensorLikeType:
+    assert isinstance(a, TensorLike)
+
+    if posinf is None:
+        posinf = prims.maximum_value(a.dtype)
+
+    if neginf is None:
+        neginf = prims.minimum_value(a.dtype)
+
+    result = where(isnan(a), nan, a)
+
+    is_neg = signbit(a)
+    is_neginf = bitwise_and(isinf(a), is_neg)
```

Comment:
cc @zasdfgbnm, who was thinking about the modeling of infinite values the other day, too